src/domain/crowdaction/interface/badge-config.interface.ts

```diff
@@ -0,0 +1,3 @@
+export interface IBadgeConfig {
+    readonly diamondThreshold: number | IBadgeConfig;
```

Comment:
I think the tests are failing because of this? You just need it to be of type `number`.